packages/connect-examples/native-android-example/app/src/main/java/com/onekey/hardware/hardwareexample/MainActivity.kt

```
@@ -0,0 +1,887 @@
+        webview = findViewById(R.id.webview)
+        
+        // Do not restore previously selected device; show only real-time scan results
+        
+        configureWebView()
+        loadHtmlFile()
+        registerHandlers()
+
+        // Rely on system pairing dialog when bonding is required
+    }
+
+    fun registerHandlers() {
```

detekt (1.23.8)

[warning] The function registerHandlers appears to be too complex based on Cognitive Complexity (complexity: 48). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)